ARCHITECTURE.md

```diff
@@ -0,0 +1,3386 @@
+        match event {
+            GameEvent::CombatAbilityUsed { source, target, ability } => {
+                format!("Entity {} used ability '{}' on entity {}",
+                    source.0, ability, target.0)
+            }
+            // ... other event types
+            _ => format!("{:?}", event),
+        }
+    }
+
+    fn extract_world_context(&self, world_state: &WorldState, _event: &GameEvent) -> WorldContext {
+        WorldContext {
+            time_of_day: format!("{:02}:{:02}", world_state.time.hour, world_state.time.minute),
+            day: world_state.time.day,
+            season: world_state.time.season,
+            weather: world_state.environment.weather,
+            is_night: world_state.is_night(),
+        }
+    }
+
+    fn extract_character_context(&self, world_state: &WorldState, event: &GameEvent) -> Vec<CharacterContext> {
+        event.involved_entities()
+            .iter()
+            .filter_map(|id| world_state.get_character(*id))
+            .map(|c| CharacterContext {
+                name: c.name.clone(),
+                title: c.title.clone(),
+                current_hp_percent: (c.stats.current_hp as f32 / c.stats.max_hp as f32 * 100.0) as u32,
+                active_statuses: c.status_effects.active_effects
+                    .iter()
+                    .map(|e| format!("{:?}", e.effect_type))
+                    .collect(),
+                personality: c.personality_traits.clone(),
+            })
+            .collect()
+    }
+}
+
+/// The assembled context ready for prompt generation
+#[derive(Debug, Clone)]
+pub struct AssembledContext {
+    pub event_description: String,
+    pub relevant_facts: Vec<String>,
+    pub world_context: WorldContext,
+    pub character_context: Vec<CharacterContext>,
+    pub activated_tags: Vec<(Tag, f32)>,
+}
+
+#[derive(Debug, Clone)]
+pub struct WorldContext {
+    pub time_of_day: String,
+    pub day: u32,
+    pub season: Season,
+    pub weather: Weather,
+    pub is_night: bool,
+}
+
+#[derive(Debug, Clone)]
+pub struct CharacterContext {
+    pub name: String,
+    pub title: Option<String>,
+    pub current_hp_percent: u32,
+    pub active_statuses: Vec<String>,
+    pub personality: Vec<String>,
+}
+```
+
+**Spreading Activation Visualization:**
+
+```mermaid
+flowchart TD
+    subgraph "Step 1: Trigger"
+        E[GameEvent: CombatAbilityUsed]
+        E --> T1[Villain_A: 1.0]
+        E --> T2[Priestess_B: 1.0]
+        E --> T3[Magic: 1.0]
+    end
+
+    subgraph "Step 2: First Spread"
+        T1 -->|0.9 × 0.5| T4[Orphanage: 0.45]
+        T2 -->|0.9 × 0.5| T4
+        T1 -->|0.7 × 0.5| T3
+        T2 -->|0.8 × 0.5| T3
+    end
+
+    subgraph "Step 3: Accumulate"
+        T4 --> ACC[Orphanage: 0.9]
+        T3 --> ACC2[Magic: 1.75]
+    end
+
+    subgraph "Step 4: Collect Facts"
+        ACC --> F1[Fact: Grew up together]
+        ACC2 --> F2[Fact: Both have magical abilities]
+    end
+```
```

Comment:
investigate this, i think it must be reversed (to build ai context), for example
for example we encounter with knights in the forest (and we must provide reference to politic situation - for example we are hiding from crown - this system should helps us to provide situation-dependent context)